pkg/security/x509source.go

```diff
@@ -156,7 +157,7 @@ func newX509Source(ctx context.Context, clock clock.Clock, cptd spiffeid.TrustDo
 		appID:          opts.AppID,
 		appNamespace:   ns,
 		trustDomain:    trustDomain,
-		kubernetesMode: os.Getenv("KUBERNETES_SERVICE_HOST") != "",
+		kubernetesMode: opts.Mode == modes.KubernetesMode,
```

Comment:
Is this a breaking change? Shouldn't we use both flags to enable this (via an OR)?

Reply:
As discussed in the call yesterday, we agreed that the `KUBERNETES_SERVICE_HOST` environment variable is fallible so instead we are determining the mode via a passed option. For Daprd, this option is coming from the existing `--mode` CLI flag. For placement, this is coming from a new CLI flag `--mode`.

it's not a breaking change.